tests/graph/test_hydrate.py

```diff
@@ -52,6 +53,31 @@ class Tree(BaseModel):
     branches: list[Branch]
 
 
+class Caterpillar(BaseModel):
+    home: Leaf | None
+
+
+@pytest.mark.parametrize(
+    ("model", "attribute", "expected"),
+    [
+        (Tree, "branch", Branch),
+        (Branch, "leaves", Leaf),
+        (Leaf, "color", "cannot hydrate paths with non base models"),
+        (Leaf, "veins", "cannot hydrate paths with non base models"),
+        (Caterpillar, "home", Leaf),
+    ],
```

Comment:
```suggestion
        (Nature, "flora_or_fauna", "Expected one non-None type, got 2"),
    ],
```

Reply:
Could you elaborate a bit? What should the `Nature` class look like?

Reply:
added in 964bdfe22fe8646eeaa63f1e47b983c5e4f11d28